src/utils/api.js

```diff
@@ -1,24 +1,78 @@
+import ArticleDates from "../utils/getDates";
+
 export default class Api {
-  addArticle = (article) => {
-    return new Promise((resolve) => {
-      resolve({
-        urlToImage: article.urlToImage,
-        title: article.title,
-        description: article.description,
-        source: article.source,
-        publishedAt: article.publishedAt,
-        url: article.url,
-        keyword: article.keyword,
-        id: article.id,
-      });
-    });
+  constructor({ baseUrl }) {
+    this._baseUrl = baseUrl;
+    this._articleDates = new ArticleDates();
+    this._pageSize = 100;
+  }
+
+  getNewsArticles(query) {
+    if (query) {
+      return fetch(`${this._baseUrl}/articles/search/${query}`, {
+        method: "GET",
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+        },
+      }).then((res) => this._checkReponse(res));
+    }
+  }
+
+  getUser = (token) => {
+    return fetch(`${this._baseUrl}/users/`, {
+      method: "GET",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${token}`,
+      },
+    }).then((res) => this._checkReponse(res));
+  };
+
+  getArticles = (token) => {
+    return fetch(`${this._baseUrl}/articles/`, {
+      method: "GET",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${token}`,
+      },
+    }).then((res) => this._checkReponse(res));
   };
 
-  removeArticle = (id) => {
-    return new Promise((resolve) => {
-      resolve({
-        id: id,
-      });
-    });
+  saveArticle = (article, token) => {
+    return fetch(`${this._baseUrl}/articles/`, {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${token}`,
+      },
+      body: JSON.stringify(article),
+    }).then((res) => this._checkReponse(res));
   };
+
+  removeArticle = (articleId, token) => {
+    return fetch(`${this._baseUrl}/articles/${articleId}`, {
+      method: "DELETE",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${token}`,
+      },
+    }).then((res) => this._checkReponse(res));
+  };
+
+  _checkReponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+
+    return res
+      .json()
+      .then((err) => {
+        return Promise.reject(`Error: ${res.status} - ${err.message}`);
+      })
+      .catch((err) => alert(err));
```

Comment:
`    .catch((err) => alert(err));`


Please, delete it

You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in  (`App.js`) after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.